packages/commonwealth/server/migrations/20240509191817-create-tags.js

```diff
@@ -0,0 +1,29 @@
+'use strict';
+
+module.exports = {
+  up: async (queryInterface, Sequelize) => {
+    return queryInterface.sequelize.transaction(async (t) => {
+      return queryInterface.createTable(
+        'Tags',
+        {
+          id: {
```

Comment:
Normally we use natural keys for the primary key (In this case it would be name).

Usually we only use surrogate keys (auto incrementing id) when there is a possibility of changing the natural key (For example if we want to change the name of a tag). But in this case we probably have no intention of adding a route to change the tag name, becuase when user's choose a tag to associate with, they don't expect it to change.

The benefit of natural keys as primary keys are that there is one less index for the db to maintain, as well as if we ever need to migrate to a sharding (distributed) solution, there will be no primary key conflicts.

Reply:
Hmm, I like the idea here but I am a little unsure. 

I saw an instance (in https://github.com/hicommonwealth/commonwealth/issues/6647) where tags had emojis in them (which we are not doing ATM), so maybe in the future we may want to add emojis to existing tag names. Will confirm from the product team if that is a possibility. 

Also, would you say that this is a blocking concern? If no, i would actually like to make a followup to get the basic feature shipped.

Reply:
So I discussed with @HIM92 and based on that, we would want to have the possibility of changing a tag name in the future.

Reply:
I think you should clarify further. Its not about adding or removing a tag name, it's more like this workflow:

Say we have a tag `Gaming` and a Profile chooses this tag. Would we want in the future to rename Gaming to some other name and to update it for each profile/community preference? Personally I don't think so, as when you select a Tag, that is your preference. Changing the name for that preference could change whether the user would have chosen that preference in the first place.

Anyways if we really do intend to change the tag name underneath the user (Not just removing the tag and adding a new one), then I guess it is fine to use the surrogate key.

Reply:
[Reply](https://github.com/hicommonwealth/commonwealth/pull/7824#issuecomment-2110552252) from Harris

cc: @kurtisassad

Reply:
Per discussion on slack, we can keep the current schema. Resolving.